### PR TITLE
extend ```multi_trigger_auto_complete``` under ```acter_trigger_auto_complete```.

### PR DESCRIPTION
**Pull request by gtalha07**
This brings the auto_complete_trigger to be included in our main repo source and a new feature to be added in regards of trigger characters. As integrating in the whole package brings large-set of files (mostly which are from base project ), should be __***BEST REVIEWED PER COMMIT***__.

### PREVIEW:

https://github.com/user-attachments/assets/fc61e669-93d6-469e-adf8-cf36143564e1

### CODE REVIEW

File: app/packages/acter_trigger_auto_complete/asset/emoji_demo.gif
Comment (gnunicorn) [inline app/packages/acter_trigger_auto_complete/asset/emoji_demo.gif:1]: Can't we collapse the removing of the demo gifs and this one into one? Thus way we are not adding the `gifs` in the first place.

This commit already contains other changes to the code itself like the renaming and the changelog, let's also remove the gifs already. I'd prefer they never made it into the repo ;) .

File: app/lib/features/chat/widgets/custom_input.dart [hunk 1]
```diff
@@ -196,33 +196,55 @@ class _ChatInput extends ConsumerStatefulWidget {
...
 
 class __ChatInputState extends ConsumerState<_ChatInput> {
-  TextEditingController controller = TextEditingController();
+  late ActerTextController textController;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _setController();
+  }
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:205]: shouldn't this be a `didWidgetChange`?

File: app/lib/features/chat/widgets/custom_input.dart [hunk 2]
```diff
@@ -196,33 +196,55 @@ class _ChatInput extends ConsumerStatefulWidget {
...
+        color: Theme.of(context).colorScheme.onSecondary,
+        height: 0.5,
+        background: Paint()
+          ..color = Theme.of(context).colorScheme.secondary
+          ..strokeWidth = 10
+          ..strokeJoin = StrokeJoin.round
+          ..style = PaintingStyle.stroke,
+      ),
+    };
+    textController = ActerTextController(triggerStyles: triggerStyles);
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:219]: why not do that in the class directly?

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+
+/// Extended [TextEditingController] which takes up trigger styles as optional.
+/// Styles the trigger inputs based on map for mentions, hashtags or emojis etc.
+/// If provided empty, will
```
Comment (gnunicorn) [inline app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart:5]: will ...?

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+class ActerTextController extends TextEditingController {
+  ActerTextController({String? text, this.triggerStyles}) : super(text: text);
```
Comment (gnunicorn) [inline app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart:7]: I think calling this `ActerTextController` is a bit weird. either call it `ActerTriggerAutoCompleteTextController` or just `TriggerAutocompleteTextController`?!?

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart [hunk 5]
```diff
@@ -25,7 +25,7 @@ class ActerTextController extends TextEditingController {
       final trigger = entry.key;
       final triggerStyle = entry.value;
 
-      final regex = RegExp('\\$trigger\\S+(?: \\S+)*');
+      final regex = RegExp('\\$trigger\\S+(?: \\S+)*(?=\\s)');
```
Comment (gnunicorn) [inline app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart:28]: I'd really like some unit tests for the regexp to understand better what to expect from it... and also understand changes made and why...

**Review by gnunicorn — CHANGES_REQUESTED** [review]
Thanks heaps. In particular for cleaning up the history, that makes it really easy to follow what needs reviews and what doesn't. Unfortunately this history still leaves the gifs and other pointless stuff in the history of our repo (which I'd like to avoid), so I have to ask you to squash the first three commits into one and rebase the rest on top of that ... Other than that, just minor inquiries to be considered.

File: app/lib/features/chat/widgets/custom_input.dart (hunk 2 above)
Comment (gtalha07) [reply app/lib/features/chat/widgets/custom_input.dart:219]: That'd mean we are taking away the customization support from the plugin, which is what it needs to offer...

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart (hunk 3 above)
Comment (gtalha07) [reply app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart:5]: Sorry, missed the comment here. Needs correction! 👍🏻

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart (hunk 4 above)
Comment (gtalha07) [reply app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart]: First one sounds better with more meaningful to plugin. Will update it!

File: app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart (hunk 5 above)
Comment (gtalha07) [reply app/packages/acter_trigger_auto_complete/lib/src/acter_text_controller.dart:28]: Ahh should've it documented explaining why specifically it's used. Also I will think of some basic tests to include hereafter.

**Review by gnunicorn — APPROVED** [review]

**Pull request closed without merging**